### PR TITLE
Update examples for sycl-nightly/20211108

### PR DESCRIPTION
**Pull request by joeatodd**
Updated the examples to build with DPC++ nightly build from 8th November 2021. 

Mostly find & replace changed names etc, but some functional changes too.

Should it really be necessary to `cudaDeviceSynchronize()` within a `host_task` before returning? Is `q.wait()` not capable of handling this? I can see that it might be difficult to implement that properly. If this is expected with USM, is it also expected with buffers?

### CODE REVIEW

File: example-02/build.sh [hunk 1]
```diff
@@ -0,0 +1,3 @@
+rm -rf build && mkdir build && cd build
+cmake ../ -DSYCL_ROOT=${SYCL_ROOT_DIR} -DCMAKE_CXX_COMPILER=${SYCL_ROOT_DIR}/bin/clang++
+make -j 8
```
Comment (Ruyk) [inline example-02/build.sh:3]:
```suggestion
make -j
```

File: example-03/vec_add_usm.cu [hunk 2]
```diff
@@ -64,6 +60,7 @@ int main(int argc, char *argv[]) {
 
         // Execute the kernel
         vecAdd<<<gridSize, blockSize>>>(d_A, d_B, d_C, n);
+        cudaDeviceSynchronize();
```
Comment (Ruyk) [inline example-03/vec_add_usm.cu:63]: A device synchronize isan overkill here, you can enqueue a SYCL async barrier after the submit and wait for that event. That should be enough. 
Alternatively, you can record a cuda event for the vecAdd kernel and wait for that one.
For the purpose of the example being self-illustrative, is good to have a code comment before the device synchronize or whatever alternative method you use explaining why is needed

File: exercise-01/build.sh [hunk 3]
```diff
@@ -0,0 +1,5 @@
+#!/bin/bash
+
+mkdir build && cd build
+cmake ../ -DSYCL_ROOT=${SYCL_ROOT_DIR} -DCMAKE_CXX_COMPILER=${SYCL_ROOT_DIR}/bin/clang++
+make -j 8
```
Comment (Ruyk) [inline exercise-01/build.sh:5]:
```suggestion
make -j
```

**Review by Ruyk — APPROVED** [review]
Thanks for updating the examples! Few minor comments and clarifications above. I haven't realised the host task implementation in DPC++ was completed, that is good news.

File: example-02/sycl_sgemm.cpp [hunk 4]
```diff
@@ -83,7 +80,7 @@ int main() {
       auto d_B = b_B.get_access<sycl::access::mode::read>(h);
       auto d_C = b_C.get_access<sycl::access::mode::write>(h);
 
-      h.codeplay_host_task([=](sycl::interop_handle ih) {
+      h.host_task([=](sycl::interop_handle ih) {
```
[the remark this answers is not in the document]
Comment (joeatodd) [reply example-02/sycl_sgemm.cpp:83]: Interesting, I had assumed that `codeplay_host_task` became `host_task` and that's why it's absent. Can you give me a 1 sentence overview of the difference? Both seem to support interop.

[the remark this answers is not in the document]
Comment (Ruyk) [reply example-02/sycl_sgemm.cpp:83]: SYCL 2020 host task uses specific accessor target modes (an additional accessor template parameter) to indicate the dependency is on the host and not the device.

**Pull request merged**